src/fetch/orderbook.py

```diff
@@ -88,7 +88,13 @@ def get_order_rewards(cls, block_range: BlockRange) -> DataFrame:
 
         # Solvers do not appear in both environments!
         assert set(prod.solver).isdisjoint(set(barn.solver)), "solver overlap!"
-        return pd.concat([prod, barn])
+        if not prod.empty and not barn.empty:
+            return pd.concat([prod, barn])
+        if not prod.empty:
+            return prod.copy()
+        if not barn.empty:
+            return barn.copy()
+        return pd.DataFrame()
```

Comment:
Looks like it does what it is supposed to. The following looks more pythonic, but I have not tested it.
```suggestion
        return pd.concat(df for df in [prod, barn] if not df.empty)
```

Reply:
Yeah agreed but i don't feel that comfortable with all these compact pythonic ways, so if you don't mind, i want to test it as is and then we can potentially make it nicer in the future.